File: frontend/src/types/analysis.ts
export type Direction = 'bullish' | 'neutral' | 'bearish';
export type TimeHorizon = 'short_term' | 'medium_term' | 'long_term';
export type Impact = 'positive' | 'negative' | 'mixed';
export type Severity = 'low' | 'medium' | 'high' | 'critical';
export type Sentiment = 'risk_on' | 'risk_off' | 'mixed';

export interface Opportunity {
  name: string;
  direction: Direction;
  confidence: number;
  time_horizon: TimeHorizon;
  reasons: string[];
  risks: string[];
  summary: string;
}

export interface MacroDriver {
  title: string;
  impact: Impact;
  summary: string;
}

export interface RiskItem {
  title: string;
  severity: Severity;
  category: string;
  description: string;
}

export interface MarketOverview {
  bullish_count: number;
  bearish_count: number;
  neutral_count: number;
  strongest_signal: string;
  highest_risk: string;
  market_sentiment: Sentiment;
  summary: string;
}

export interface AnalysisResponse {
  generated_at: string;
  overview: MarketOverview;
  top_opportunities: Opportunity[];
  macro_drivers: MacroDriver[];
  risks: RiskItem[];
}

File: backend/app/schemas/analysis.py
from pydantic import BaseModel
from typing import Literal
from datetime import datetime


class Opportunity(BaseModel):
    name: str
    direction: Literal["bullish", "neutral", "bearish"]
    confidence: int
    time_horizon: Literal["short_term", "medium_term", "long_term"]
    reasons: list[str]
    risks: list[str]
    summary: str


class MacroDriver(BaseModel):
    title: str
    impact: Literal["positive", "negative", "mixed"]
    summary: str


class RiskItem(BaseModel):
    title: str
    severity: Literal["low", "medium", "high", "critical"]
    category: str
    description: str


class MarketOverview(BaseModel):
    bullish_count: int
    bearish_count: int
    neutral_count: int
    strongest_signal: str
    highest_risk: str
    market_sentiment: Literal["risk_on", "risk_off", "mixed"]
    summary: str


class AnalysisResponse(BaseModel):
    generated_at: datetime
    overview: MarketOverview
    top_opportunities: list[Opportunity]
    macro_drivers: list[MacroDriver]
    risks: list[RiskItem]


class AnalyzeRequest(BaseModel):
    focus_sectors: list[str] | None = None
    time_horizon: Literal["short_term", "medium_term", "long_term"] | None = None
